dev/error_message_refiner.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python3
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Utility for refining error messages based on LLM.
+
+Usage:
+    python error_message_refiner.py <error_class> [--gpt_version=<version>]
+
+Arguments:
+    <error_class>           Required.
+                            The name of the error class to refine the messages for.
+                            The list of error classes is located in
+                            `core/src/main/resources/error/error-classes.json`.
+
+Options:
+    --gpt_version=<version> Optional.
+                            The version of Chat GPT to use for refining the error messages.
+                            If not provided, the default version("gpt-3.5-turbo") will be used.
+
+Example usage:
+    python error_message_refiner.py CANNOT_DECODE_URL --gpt_version=gpt-4
+
+Description:
+    This script refines error messages using the LLM based approach.
+    It takes the name of the error class as a required argument and, optionally,
+    allows specifying the version of Chat GPT to use for refining the messages.
+
+    Options:
+        --gpt_version: Specifies the version of Chat GPT.
+                       If not provided, the default version("gpt-3.5-turbo") will be used.
+
+    Note:
+    - Ensure that the necessary dependencies are installed before running the script.
```

Comment:
What are the dependencies?

Reply:
We need `openai` (https://github.com/openai/openai-python) for running this script. Maybe should we add this to the `dev-requirements.txt` ??